2.7/Dockerfile

```diff
@@ -44,16 +44,16 @@ RUN set -ex \
 	&& make install \
 	&& ldconfig \
 	\
-		&& wget -O /tmp/get-pip.py 'https://bootstrap.pypa.io/get-pip.py' \
-		&& python2 /tmp/get-pip.py "pip==$PYTHON_PIP_VERSION" \
-		&& rm /tmp/get-pip.py \
+	&& wget -O /tmp/get-pip.py 'https://bootstrap.pypa.io/get-pip.py' \
+	&& python2 /tmp/get-pip.py "pip==$PYTHON_PIP_VERSION" \
+	&& rm /tmp/get-pip.py \
 # we use "--force-reinstall" for the case where the version of pip we're trying to install is the same as the version bundled with Python
 # ("Requirement already up-to-date: pip==8.1.2 in /usr/local/lib/python3.6/site-packages")
 # https://github.com/docker-library/python/pull/143#issuecomment-241032683
 	&& pip install --no-cache-dir --upgrade --force-reinstall "pip==$PYTHON_PIP_VERSION" \
-# then we use "pip list" to ensure we don't have more than one pip version installed
+# then we use "pip freeze" to ensure we don't have more than one pip version installed
 # https://github.com/docker-library/python/pull/100
-	&& [ "$(pip list |tac|tac| awk -F '[ ()]+' '$1 == "pip" { print $2; exit }')" = "$PYTHON_PIP_VERSION" ] \
+	&& [ "$(pip freeze --all |tac|tac| awk -F '==' '$1 == "pip" { print $2; exit }')" = "$PYTHON_PIP_VERSION" ] \
```

Comment:
Very nice -- `pip freeze` is indeed a much better solution. :+1:

The `|tac|tac|` is a cheeky trick to ensure the entire output of `pip freeze` is read completely -- some applications get really grumpy when we do `exit` from `awk` too quickly and standard input is then closed (causing `pip freeze` to have standard output closed).  It's easier to illustrate with `curl`+`head`:

```console
$ curl -fsSL 'https://google.com' | head -1 > /dev/null
curl: (23) Failed writing body (867 != 1339)
```